### PR TITLE
feat(transparency)!: add decoupled transparency options;

### LINKED ISSUE #126: Feat: Background only in transparent mode
When enabling transparent mode, we should provide an option where only the background is transparent, and other floating windows and components are opaque.

### PR DESCRIPTION
**Pull request by 5-pebbles**
Closes #126 

You can now set the transparency of the background & floating windows separately.

The settings are now in the table transparent.
Here are the default values:

```lua
transparent = {
    bg = false,
    float = false,
},
````

BREAKING CHANGE: The `transparent_bg` option has been removed;

### CODE REVIEW

**Comment by 5-pebbles** [comment]
I think leaving a deprecated warning for a few weeks is a good idea.

**Comment by AlexvZyl** [comment]
> I think leaving a deprecated warning for a few weeks is a good idea.

I think we can just internally keep the old value for backwards compatibility.  I really do not want to break anyone's setup.

**Comment by 5-pebbles** [comment]
Done & I will do the same for the on_highlight option for #124

**Comment by AlexvZyl** [comment]
@5-pebbles Take a look at `:h nvim_notify`.  We can use that to notify the user that he is using an outdated option.

**Comment by 5-pebbles** [comment]
So that would make it forward to `noice.nvim` not just echo it, should I add the warning back in using that?

**Comment by AlexvZyl** [comment]
> So that would make it forward to `noice.nvim` not just echo it, should I add the warning back in using that?

It will forwrad to noice if it is setup, otherwise it will just be a normal print I believe.

**Comment by 5-pebbles** [comment]
So, should I add the deprecated warning back in using `nvim_notify`?

File: lua/nordic/config.lua [hunk 1]
```diff
@@ -52,6 +57,15 @@ M.options = defaults
 
 -- called automatically by load
 function M.setup(options)
+    if options.transparent_bg then
```
Comment (AlexvZyl) [inline lua/nordic/config.lua:60]: Remember to call `nvim_notify`

**Review by AlexvZyl — CHANGES_REQUESTED** [review]

**Comment by AlexvZyl** [comment]
> So, should I add the deprecated warning back in using `nvim_notify`?

Ah sorry, yes please.

**Comment by AlexvZyl** [comment]
Also, I think we should add a file called 'compatibility.lua' where we do all the backwards compatibility logic.

**Comment by AlexvZyl** [comment]
> Also, I think we should add a file called `compatibility.lua` where we do all the backwards compatibility logic.

Or maybe a function in `config.lua`

**Comment by 5-pebbles** [comment]
> Also, I think we should add a file called `compatibility.lua` where we do all the backwards compatibility logic.

Good idea.

> Or maybe a function in `config.lua`

I think a new file is better, config.lua is mostly defaults.

**Comment by AlexvZyl** [comment]
Do you get the notification when you use the outdated config option? Does not seem to work on my side.

Also, we should consider using `vim.notify_once()`.

File: lua/nordic/compatibility.lua [hunk 2]
```diff
@@ -0,0 +1,20 @@
+local function compatability(options)
+    -- All backwards compatibility
+
+    -- Log level
+    local level = vim.log.levels.WARN
+
+    if options.transparent_bg then
+        vim.notify('Nordic.nvim: transparent_bg is deprecated, use transparent instead', level)
```
Comment (AlexvZyl) [inline lua/nordic/compatibility.lua:8]: Maybe we should use "config.transparent_bg" and "config.transparent" to make it a bit more clear?

**Review by AlexvZyl — CHANGES_REQUESTED** [review]

File: lua/nordic/compatibility.lua (hunk 2 above)
Comment (5-pebbles) [reply lua/nordic/compatibility.lua]: Definitely: [feat(deprecation): better warning message;](https://github.com/AlexvZyl/nordic.nvim/pull/127/commits/79bd1b54e318c1b2f4ef744551025e46d2d7251b)

**Comment by AlexvZyl** [comment]
Also, we should do this:

```lua
vim.notify_once("...", level, {title = "nordic.nvim"})
```

**Comment by 5-pebbles** [comment]
> Also, you can add the title to the notification message:
> 
> ```lua
> vim.notify_once("...", level, {title = "nordic.nvim"})
> ```
As far as I can tell that does not effect the normal echo.  So I am going to leave the title in the message as well.

**Comment by AlexvZyl** [comment]
The last thing I think we should do is change the notification setup based on if notify and/or noice is setup or not.  But I can also do that.

**Comment by AlexvZyl** [comment]
> The last thing I think we should do is change the notification based on if notify and/or noice is setup or not. But I can also do that.

Nevermind I think this is unnecessary.

**Comment by AlexvZyl** [comment]
@5-pebbles Please check my latest changes, then I think we can merge.

File: lua/nordic/colors/init.lua [hunk 3]
```diff
@@ -38,12 +38,12 @@ function C.extend_palette()
         options.cursorline.bg = C.black0
     end
 
-    C.bg_visual = (options.transparent_bg and options.cursorline.bg)
+    C.bg_visual = ((options.transparent.bg or options.transparent.float) and options.cursorline.bg)
```
[the remark this answers is not in the document]
Comment (5-pebbles) [reply lua/nordic/colors/init.lua]: You are correct I just fixed it.

**Comment by 5-pebbles** [comment]
Everything looks good!

[Ref(compat): Remove occurnces of transparent_bg](https://github.com/AlexvZyl/nordic.nvim/pull/127/commits/bdf216d0a294023e08460e0a647aba5d06ad7cb5) thanks for noticing this! I need to check my merges more carefully...

**Pull request merged**